[Reopen] Let promise rejection propagate in handleSubmit

JoseLion opened the pull request:
Resolve #2758

Reopening #3052

BrOrlandi commented:
Hi!
This issue seems very important to me, as @hixus said, it prevents errors from being captured on Sentry or similar.
@jaredpalmer could you check this?

avasuro commented:
+1 to previous comments. This part of code doesn't allow to catch errors raised in onSubmit handler using `window.onunhandledrejection`.
@jaredpalmer, please, take a look

probablyup commented:
If you're still around, could you add a changeset for this? Or someone can cherry pick into a new PR and add one. Thanks!

JoseLion commented:
@probablyup I updated the branch and added a chageset following [these instructions](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md).  Let me know if something's wrong or missing 🙂
